test/unit/kubernetes-deploy/kubernetes_resource/deployment_test.rb

```diff
@@ -366,12 +367,13 @@ def build_rs_template(status: { 'replicas' => 3 })
     result
   end
 
-  def build_synced_deployment(template:, replica_sets:)
+  def build_synced_deployment(template:, replica_sets:, expect_pod_get: nil)
```

Comment:
why `nil` seems like `true` is a better fit here

Reply:
The real default value I want is `replica_sets.present?`, and for this to be an override. That's not possible if the override's default value is `true`. The alternative is to default it to true, remove L375, and make all the callers do `expect_pod_get: replica_sets.present?` themselves.